main.py

```diff
@@ -24,9 +25,60 @@
 |   7   |   8   |   9   |
 |       |       |       |
 +-------+-------+-------+'''
+
+listBoard = list(board)
+print(board)
+
 def userMove():
     userInput = input("place O where: ")
-    x = list(board)
-    x[boardIndex[userInput]] = "O"
-    newString = "".join(x)
-    print(newString)
+
+    listBoard[boardIndex[userInput]] = "O"
+
+    global updatedBoard
+    updatedBoard = "".join(listBoard)
+    print(updatedBoard)
+
+    if checkVictory("human") == "Player":
+        print("Player Wins!")
+    else:
+        computerMove()
+
+def computerMove():
+    computerInput = random.randint(1,9)
+
+    listBoard[boardIndex[str(computerInput)]] = "X"
+    updatedBoard = "".join(listBoard)
+    print(updatedBoard)
+
+    if checkVictory("computer") == "computer":
+        print("Computer Wins!")
+    else:
+        userMove()
+          
+
+def checkVictory(player):
+    currentBoard = []
+
+    #updating board to latest info
+    for position in range(1,10):
+        currentBoard.append(board[boardIndex[str(position)]])
+
+    if player == "human": 
+
+        #only 4 win conditions
+        if currentBoard[0] == "O" and currentBoard[1] == "O" and currentBoard[2] == "O" or currentBoard[6] == "O" and currentBoard[7] == "O" and currentBoard[8] == "O": 
+            return "Player"
+        elif currentBoard[0] == "O" and currentBoard[3] == "O" and currentBoard[6] == "O" or currentBoard[2] == "O" and currentBoard[5] == "O" and currentBoard[8] == "O":
+            return "Player"
+        else:
+            return "NoWin"
+
+    elif player == "computer": #must check all 8 win conditions because players first move is not a constant
```

Comment:
did not check diagonal solutions. "and" statements may be ruining the functionality of the code. do more testing.

Reply:
missing function to check if its a draw. players can overlap already existing moves.